pkg/volume/util/resize_util.go

```diff
@@ -220,8 +220,13 @@ func MarkFSResizeFinished(
 
 	// if RecoverVolumeExpansionFailure is enabled, we need to reset ResizeStatus back to nil
 	if utilfeature.DefaultFeatureGate.Enabled(features.RecoverVolumeExpansionFailure) {
-		expansionFinished := v1.PersistentVolumeClaimNoExpansionInProgress
-		newPVC.Status.ResizeStatus = &expansionFinished
+		allocatedResourceStatusMap := newPVC.Status.AllocatedResourceStatuses
```

Comment:
Can unit tests be added for this?

Reply:
Added whole bunch of unit tests for various resize functions.